internal/provider/resource_application_test.go

```diff
@@ -669,6 +836,53 @@ resource "juju_application" "{{.AppName}}" {
 		})
 }
 
+func testAccResourceApplicationWithCustomResources(modelName, channel string, resourceName string, customResource string) string {
+	return fmt.Sprintf(`
+resource "juju_model" "this" {
+  name = %q
+}
+
+resource "juju_application" "this" {
+  model = juju_model.this.name
+  name = "test-app"
+  charm {
+    name     = "sdcore-ausf-k8s"
```

Comment:
s/sdcore-ausf-k8s/grafana-k8s/ will fix the tests.

Reply:
Hello @hmlanigan, 

Integration tests and static analysis and linting issues are fixed. 
It's ready for re-review. 
Many thanks!